src/predictor/gpu_predictor.cu

```diff
@@ -352,40 +310,45 @@ class GPUPredictor : public xgboost::Predictor {
     InitModel(model, tree_begin, tree_end);
 
     size_t batch_offset = 0;
+    auto* preds = out_preds;
+    std::unique_ptr<HostDeviceVector<bst_float>> batch_preds{nullptr};
```

Comment:
instead of dual copy: 
- copying input prediction vector into a temp one based on batch size and, 
- copying back the output prediction vector from the kernel back into `out_preds` 

is it not possible to:
-  slice and dice the `out_preds` based on batch size - basically, get a device span from `out_preds` and get a subspan based on the batch size and batch offset
- pass the subspan to predict internal as a pass through to the kernel

Reply:
That'll assume the whole predictor vector will be on device, right? I think the current approach doesn't put the whole vector on device in external memory mode, although I haven't 100% verified it.

Reply:
even if the device copy is prevented here, it'll be copied right back (soon after this) in the objective functions while computing the gradients, or later when the prediction cache is updated during training.